reducers/views/rhs.js

```diff
@@ -57,19 +68,71 @@ function fromFlaggedPosts(state = false, action) {
 function fromPinnedPosts(state = false, action) {
     switch (action.type) {
     case ActionTypes.SELECT_POST:
-        if (action.from_pinned_posts) {
-            return action.from_pinned_posts;
+        if (action.fromPinnedPosts) {
+            return action.fromPinnedPosts;
         }
         return false;
+    case ActionTypes.UPDATE_RHS_STATE:
+        return false;
+    default:
+        return state;
+    }
+}
+
+function fromMentions(state = false, action) {
+    switch (action.type) {
+    case ActionTypes.SELECT_POST:
+        if (action.fromMentions) {
+            return action.fromMentions;
+        }
+        return false;
+    case ActionTypes.UPDATE_RHS_STATE:
+        return false;
+    default:
+        return state;
+    }
+}
+
+function rhsState(state = null, action) {
+    switch (action.type) {
+    case ActionTypes.UPDATE_RHS_STATE:
+        return action.state;
+    case ActionTypes.SELECT_POST:
+        return null;
+    default:
+        return state;
+    }
+}
+
+function searchTerms(state = '', action) {
+    switch (action.type) {
+    case ActionTypes.UPDATE_RHS_SEARCH_TERMS:
+        return action.terms;
+    default:
+        return state;
+    }
+}
+
+function isSearching(state = false, action) {
+    switch (action.type) {
+    case SearchTypes.SEARCH_POSTS_REQUEST:
+        return true;
+    case SearchTypes.SEARCH_POSTS_FAILURE:
+    case SearchTypes.SEARCH_POSTS_SUCCESS:
+        return false;
     default:
         return state;
     }
 }
 
 export default combineReducers({
     selectedPostId,
-    selectedPostChannelId,
+    selectedChannelId,
     fromSearch,
     fromFlaggedPosts,
-    fromPinnedPosts
+    fromPinnedPosts,
+    fromMentions,
```

Comment:
Hmm, not a huge fan of having a reduce for each of these: fromSearch, fromFlaggedPosts, fromPinnedPosts, fromMentions

The only real purpose served by these is so that we can go back to the previous RHS state and only one of them should ever be true at a time. Why not replace all these with a single reducer `prevRhsState`? What do you think?

Reply:
This is the way it was before and I didn't think to change it. But I agree all that state having to do with a selected post should be grouped together. Makes things look nicer.

Shouldn't be too much effort to clean that up 👍